crates/trie/sparse/src/trie.rs

```diff
@@ -618,15 +618,15 @@ impl RevealedSparseTrie {
 
     fn rlp_node(&mut self, path: Nibbles, prefix_set: &mut PrefixSet) -> RlpNode {
         // stack of paths we need rlp nodes for
-        let mut path_stack = Vec::from([path]);
+        let mut path_stack = Vec::from([(path, false)]);
         // stack of rlp nodes
         let mut rlp_node_stack = Vec::<(Nibbles, RlpNode)>::new();
         // reusable branch child path
         let mut branch_child_buf = SmallVec::<[Nibbles; 16]>::new_const();
         // reusable branch value stack
         let mut branch_value_stack_buf = SmallVec::<[RlpNode; 16]>::new_const();
 
-        'main: while let Some(path) = path_stack.pop() {
+        'main: while let Some((path, is_in_prefix_set)) = path_stack.pop() {
```

Comment:
should this be an option instead? `is_in_prefix_set` is ambiguous because it might just mean that we haven't checked the prefix set yet